localstack/services/awslambda/lambda_executors.py

```diff
@@ -299,7 +295,7 @@ def _execute(self, func_arn, func_details, event, context=None, version=None):
         command = ''
         events_file = ''
 
-        if USE_CUSTOM_JAVA_EXECUTOR and is_java_lambda(runtime):
+        if is_java_lambda(runtime):
```

Comment:
Thanks for this change @mgagliardo . I think we need to be careful here - with this change, we're essentially enabling the custom Java executor for *all* Java Lambdas. Some time ago we switched from using our custom executor to using the default executor in [Lambci](https://github.com/lambci/docker-lambda), as it provided better compatibility.

I think we should either:
1. Check if `LAMBDA_JAVA_OPTS` is configured and only then use the custom executor, or
2. See how we can hook in our JVM flags into the `entrypoint` of the Java executor in the Lambci image

Reply:
Hello @whummer, I think for `2.` that is done from [line 707 on](https://github.com/localstack/localstack/blob/3f100e7b60a0165251db95953bfe2fa7aa93ec05/localstack/services/awslambda/lambda_executors.py#L707).

That being said, I'll go for number `1.` then.